integrated_channels/canvas/client.py

```diff
@@ -378,37 +378,65 @@ def _handle_canvas_assignment_retrieval(
             course_id (str) : the Canvas course ID relating to the course which the client is currently
             transmitting learner data to.
         """
-        # First, check if the course assignment already exists
+        # Check if the course assignment already exists
```

Comment:
a quick thought: shouldn't we increase the per_page size to larger value so we won't even need to loop in most cases? of course we still want the logic you added. Just a thought on whether simply increasing per_page will in many cases get us a hit the first time?

Reply:
I don't know what an average assignment count for a user will be, but looks like getting at least 50 or so in a single call may be worthwhile to reduce end-end calls? presuming it won't be too much to ask for a single request

Reply:
According to the Canvas documentation:

> There is an unspecified limit to how big you can set per_page to, so be sure to always check for the Link header.

As it reads, it seems to me that the limit is either purposely hidden or varies instance to instance. I'm worried about over setting the page limit and causing issues. After that, it was a matter debating what is a number that's not _**too**_ big yet big enough and from discussions with @edx-abolger, we figured that extra API calls isn't too detrimental and we know that the default size will always work with whatever instance we're integrated with.

Reply:
haha yeah that line read like per_page=1million even works lol. ok great, sounds like you have already gone over this consideration! Just wanted to ensure we thought about this given the no. of assignments returned will likely grow over time for a given course, and can be large for large enrollment size courses (bulk enrollment in Canvas lol!) 

But yeah,  we should look at real data once we launch this anyway before more changes, this setup of per_page 10 may turn out to be ideal even!